src/axes/axis.ts

```diff
@@ -105,6 +105,27 @@ export class Axis<D> extends Component {
     this._scale.offUpdate(this._rescaleCallback);
   }
 
+  /**
+   * Gets the tick label data on a element.
+   *
+   * @param {Element} element
+   */
+  public tickLabelDataOnElement(element: Element) {
+    if (element === null || element === undefined) { return undefined; }
+
+    let tickLabel: Element;
+    // go up DOM tree to find tick label element in ancestor elements
+    while ((element !== undefined) && (element.classList) && (tickLabel === undefined)) {
+      if (element.classList.contains(Axis.TICK_LABEL_CLASS)) {
+        tickLabel = element;
+      } else {
+        element = element.parentNode as Element;
+      }
+    }
+
+    return element === undefined ? element : d3.select(element).datum();
```

Comment:
nit: returning "element" to mean undefined is a little confusing; return `undefined` itself maybe?